### PR TITLE
wip: feat: Add API endpoint for favourites session of user

### PR DESCRIPTION
**Pull request by codedsun**
Fixes # https://github.com/fossasia/open-event-frontend/issues/5763

#### Short description of what this resolves:
feat: Add API endpoint for favourites session of user

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/models/user_favourite_session.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from app.models import db
+from app.models.base import SoftDeletionModel
+
+
+class UserFavouriteSession(SoftDeletionModel):
+    __tablename__ = 'user_favourite_session'
+
+    id = db.Column(db.Integer, primary_key=True)
+    session_id = db.Column(db.Integer, db.ForeignKey('sessions.id', ondelete='CASCADE'))
+    event_id = db.Column(db.Integer, db.ForeignKey('events.id', ondelete='CASCADE'))
```
Comment (iamareebjamal) [inline app/models/user_favourite_session.py:10]: Event ID not needed

File: app/api/routes.py [hunk 2]
```diff
@@ -914,6 +920,32 @@
     '/user-favourite-events/<int:id>/relationships/event',
 )
 
+# user favourite sessions
+api.route(
+    UserFavouriteSessionListPost, 'user_favourite_session_post', '/user-favourite-session'
```
Comment (iamareebjamal) [inline app/api/routes.py:925]: No need of fetching all favourites

File: app/api/routes.py [hunk 3]
```diff
@@ -914,6 +920,32 @@
...
+)
+api.route(
+    UserFavouriteSessionList,
+    'user_favourite_session_list',
+    '/user-favourite-session',
+    '/users/<int:user_id>/user-favourite-session',
```
Comment (iamareebjamal) [inline app/api/routes.py:931]: /users/<id>/favourite-sessions

Comment (iamareebjamal) [reply app/api/routes.py:931]: Another routes which need to be present:

/event/<id>/favourite-sessions: Returns favourite sessions of user in that event
/session/<id>/favourite-sessions: Returns all favourites of a session

File: app/api/routes.py (hunk 2 above)
Comment (codedsun) [reply app/api/routes.py:925]: ? It's a post request to create a favourite session

Comment (iamareebjamal) [reply app/api/routes.py:925]: OK, got it

**Comment by iamareebjamal** [comment]
Done in #7582

**Pull request closed without merging**